dashboard/lib/services/i18n/curriculum_sync_utils/serializers.rb

```diff
@@ -28,7 +23,10 @@ def serializable_hash(adapter_options = nil, options = {}, adapter_instance = se
           # default of only a single level of nesting.
           options[:include_directive] ||= JSONAPI::IncludeDirective.new('**', allow_wildcard: true)
           # compact the result, excluding not only nil values but also empty ones
-          super.reject {|_, v| v.blank?}
+          result = super.reject {|_, v| v.blank?}
+          raise KeyEror.new("Serializer must define :crowdin_key for curriculum content I18N serialization; got #{result.keys.inspect}") unless result.key?(:crowdin_key)
+          crowdin_key = result.delete(:crowdin_key)
+          {crowdin_key => result}
```

Comment:
for my understanding only: how do we treat collection serializers & serializers differently and why?

Reply:
## Single object serialization
We override the default serialization for single objects because we want the `crowdin_key` moved outside the object. For example:
```json
{
  "crowdin_key": "/s/express-2021/lessons/1/levels/2",
  "short_instructions": "In this lesson you will be..."
}
```
becomes 
```json
{
  "/s/express-2021/lessons/1/levels/2": {
    "short_instructions": "In this lesson you will be..."
  }
}
```
We want this change so that the `crowdin_key` is in the context for the String while a translator is on Crowdin.
![image](https://user-images.githubusercontent.com/1372238/163613006-a14e93d2-34b8-4746-87a8-3160579e6266.png)
## Collection of objects serialization
We override the default serialization for a collection of objects because we want collections to be `Hash`'s instead of `Array`'s. We want this because otherwise there would be unecessary array indexes in the Crowdin String Context.
| Using | Resulting Context |
| --- | ---|
| Array | script -> express-2021 -> lessons[0] -> /s/express-2021/lessons/1/levels/2 -> short_instructions |
| Hash | script -> express-2021 -> lessons -> /s/express-2021/lessons/1/levels/2 -> short_instructions |

```json
{
  "lessons": [
    {
      "/s/express-2021/lessons/1/levels/2": {
        "short_instructions": "In this lesson you will be..."
      }
    },
    {
      "/s/express-2021/lessons/1/levels/3": {}
    }
  ]
}
```
becomes

```json
{
  "lessons": {
    "/s/express-2021/lessons/1/levels/2": {
      "short_instructions": "In this lesson you will be..."
    },
    "/s/express-2021/lessons/1/levels/3": {}
  }
}
```